### ISSUE TITLE
[Feature Request] English Translation 英文翻译

### ISSUE THREAD

**Issue by linehman** [Anankke/SSPanel-UIM#2003]
**Feature 功能请求**

Describe the feature you want. 请描述你想要的功能。

Hi, Please add support for multi language. The community will help in translating the project. Believe me when I say that this project is very usefull for people from other countries as well.
Kind regards

您好，请添加对多语言的支持。社区将帮助翻译该项目。相信我，这个项目对来自其他国家的人也非常有用。
亲切的问候

**Comment by M1Screw** [comment]
Not a complex feature to add, but it would be time-consuming. Initial support can be added by 2023.5 or 2023.6 with 2-3 languages, no guarantee though.

**Comment by linehman** [comment]
Glad to hear that. there is no rush for adding this feature, since it's not a priority. That time frame is great. thank you

**Comment by M1Screw** [comment]
I just spent a few hours researching this feature, it probably requires a full template engine swap(i.e., switch to a more modern one like Twig), which means a lot of preparation work needs to be done. For example, the separation of the backend and frontend code, there is just too much fucking logic embedded in the current theme file, this needs to be changed before we can consider anything i18n.

**Comment by linehman** [comment]
I see. That's unfortunate. Hopefully it can be done some day in the future when the other issues you mentioned are taken care of.
Kind regards

**Comment by IRANWEBFA** [comment]
Please Add Persian language ❤

**Comment by M1Screw** [comment]
Update: Twig template engine has been added which enabled multi-template engine support, soon template converting and translating work can begin.
